javascript/ql/lib/semmle/javascript/frameworks/Spife.qll

```diff
@@ -303,114 +288,128 @@ module Spife {
       result = this.getAHeaderSource().getAPropertyWrite().getPropertyNameExpr().flow()
     }
 
-    override RouteHandler getRouteHandler() { result = reply.getRouteHandler() }
+    override RouteHandler getRouteHandler() { result = this.(ReplyCall).getRouteHandler() }
```

Comment:
```suggestion
    override RouteHandler getRouteHandler() { result = super.getRouteHandler() }
```

Reply:
I had to restore this cast to avoid this error `Predicate getRouteHandler/0 is ambiguous between: predicate CookieDefinition.HTTP::Http::CookieDefinition::getRouteHandler() from HTTP.qll:153, predicate ReplyCall.Spife::Spife::ReplyCall::getRouteHandler() from Spife.qll:229. You can resolve the ambiguity by qualifing super with the desired type.`

Reply:
You can do `ReplyCall.super.getRouteHandler()` in that case.

Reply:
(And similarly for the others).

Reply:
Thanks @erik-krogh, I was not aware of that syntax. What is the difference between the two?

Reply:
When a class has multiple super-classes there might be multiple incompatible `getRouteHandler()` predicates in the class hierarchy.  
With `ReplyCall.super` you specify which super-classes to look for the predicate in.